packages/react-components/react-dialog/src/utils/useDisableBodyScroll.ts

```diff
@@ -0,0 +1,56 @@
+import { useFluent_unstable } from '@fluentui/react-shared-contexts';
+import * as React from 'react';
+
+/**
+ * Hook that provides methods for enabling and disabling body scrolling through `overflow:hidden` method
+ */
+export function useDisableBodyScroll() {
+  const overflowSettingRef = React.useRef<string>();
+  const paddingRightRef = React.useRef<string>();
+  const { targetDocument } = useFluent_unstable();
+
+  const setOverflowHidden = React.useCallback(() => {
+    if (!targetDocument || !targetDocument.defaultView) {
+      return;
+    }
+    if (paddingRightRef.current === undefined) {
+      const paddingRight = computeScrollBarSizeAndPadding(targetDocument, targetDocument.defaultView);
+      if (paddingRight) {
+        paddingRightRef.current = targetDocument.body.style.paddingRight;
+        targetDocument.body.style.paddingRight = `${paddingRight}px`;
+      }
+    }
+    if (overflowSettingRef.current === undefined) {
+      overflowSettingRef.current = targetDocument.body.style.overflow;
+      targetDocument.body.style.overflow = 'hidden';
+    }
+  }, [targetDocument]);
+
+  const restoreOverflowSetting = React.useCallback(() => {
+    if (!targetDocument) {
+      return;
+    }
+    if (paddingRightRef.current !== undefined) {
+      targetDocument.body.style.paddingRight = paddingRightRef.current;
+      paddingRightRef.current = undefined;
+    }
+
+    if (overflowSettingRef.current !== undefined) {
+      targetDocument.body.style.overflow = overflowSettingRef.current;
+      overflowSettingRef.current = undefined;
+    }
+  }, [targetDocument]);
+  return { disableBodyScroll: setOverflowHidden, enableBodyScroll: restoreOverflowSetting };
+}
+
+function computeScrollBarSizeAndPadding(document: Document, view: Window) {
+  const scrollBarSize = view.innerWidth - document.documentElement.clientWidth;
+
+  if (scrollBarSize > 0) {
+    const computedBodyPaddingRight = parseInt(
+      view.getComputedStyle(document.body).getPropertyValue('padding-right'),
+      10,
+    );
+    return computedBodyPaddingRight + scrollBarSize;
+  }
```

Comment:
I don't have a proposal or an advice there (and I don't think that we can do anything), but FYI: it seems that we have a full house of things that are [causing recalc/reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a)

Reply:
Is this necessary ? is it just to avoid jumps when opening/closing the dialog ? Neither v0 nor v8 support that.

Reply:
would it be possible to do this with CSS ? https://codesandbox.io/s/strange-currying-5vswpk?file=/src/styles.css

Reply:
> would it be possible to do this with CSS ? https://codesandbox.io/s/strange-currying-5vswpk?file=/src/styles.css

Good idea! I'll check that out!

Reply:
There are two alternatives to prevent scrolling since scrolling events cannot be prevented:

1. `overflow: hidden`
2. add a bunch of listeners to prevent interactions that might cause scroll:
    - `DOMMouseScroll`
    - `mousewheel`
    - `touchmove`
    - `keydown`]

To keep the scrollbar and make it invisible with CSS we cannot use `overflow: hidden` - since it actually removes the scrollbar - which means we'd need to add a bunch of listeners to prevent events that might cause scroll (and something tells me that this is going to be hell between browsers).

So the alternatives are:

1. `overflow: hidden` + compute padding to compensate for the scrollbar removal
2. add a bunch of listeners to prevent interactions that might cause scroll + CSS that makes scrollbar invisible.

Reply:
in that case... can we just have jumps with a disappearing scrollbar until it becomes a requirement to do otherwise? 😅 

As I said both v0 and v8 do exactly that currently